app/Phragile/AssigneeCollection.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Phragile;
+
+class AssigneeCollection {
+
+	private $assignees;
+
+	public function __construct(PhabricatorAPI $phabricator, array $tasks)
+	{
+		$assigneePHIDs = $this->extractUniqueAssignees($tasks);
+		$this->assignees = $this->processAssignees(
+			$phabricator->getUserData(['phids' => $assigneePHIDs])
+		);
+	}
+
+	private function extractUniqueAssignees(array $tasks)
+	{
+		return array_unique(array_map(function($task)
+		{
+			return $task['ownerPHID'];
+		}, $tasks));
+	}
+
+	private function processAssignees(array $assigneeData)
+	{
+		$assignees = [];
+
+		foreach ($assigneeData as $assignee)
+		{
+			$assignees[$assignee['phid']] = $assignee;
+		}
+
+		return $assignees;
+	}
+
+	/**
+	 * @param string $phid
+	 * @return string
+	 */
+	public function getName($phid)
+	{
+		return isset($this->assignees[$phid]['userName']) ? $this->assignees[$phid]['userName'] : '-';
```

Comment:
I personally prefer using array_key_exists when checking if an element is there or not, as I think this is less ambiguous about its intent. Just an FYI, I do not think this needs to be changed.

The `-` that is returned strikes me as a presentation concern. What if you want to replace the `-` in the page with `<i>none</i>`? Will you change it here?